### PR TITLE
JCR-4536 optionally allow insecure TLS connections (self-signed cert,

### PR DESCRIPTION
**Pull request by kwin**
invalid common name)

### CODE REVIEW

**Comment by kwin** [comment]
This replaces #88. @reschke  Please tell me what you think!

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java [hunk 1]
```diff
@@ -341,15 +353,41 @@ public RepositoryServiceImpl(String uri, IdFactory idFactory,
             throw new RepositoryException(e);
         }
 
+        // TODO: move to ConnectionOptions as well
```
Comment (kwin) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java]: I tend to move this option as well!

Comment (kwin) [reply jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java]: Done in https://github.com/apache/jackrabbit/pull/92/commits/9ef98400bcf513aa3ce1fedabad95a032a3d543e

**Comment by kwin** [comment]
I tested the changes with https://github.com/apache/jackrabbit-filevault/pull/92/files and it works fine there.

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/ConnectionOptions.java [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+    private final int connectionTimeoutMs;
+    private final int requestTimeoutMs;
+    private final int socketTimeoutMs;
+
+    /**
+     * Boolean flag whether to use the default Java system properties for setting proxy, TLS and further options.
+     * Default = {@code false}.
+     * 
+     * @see <a href="https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/client/HttpClientBuilder.html">HttpClientBuilder</a> 
+     */
```
Comment (reschke) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/ConnectionOptions.java:52]: we should be able to use {@link ...} here

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/ConnectionOptions.java [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+    }
+
+    private static int getIntegerValueFromParameter(String parameterPrefix, Map<?, ?> parameters, int defaultValue, String... parameterKeys) {
+        for (String key : parameterKeys) {
+            Object value = parameters.get(parameterPrefix+key);
+            if (value != null) {
+                try {
+                    return Integer.parseInt(value.toString());
+                } catch ( NumberFormatException e ) {
+                    // using default
```
Comment (reschke) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/ConnectionOptions.java:331]: maybe log the exception?

(also whitespace)

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/Spi2davRepositoryServiceFactory.java [hunk 4]
```diff
@@ -136,22 +140,6 @@ public RepositoryService createRepositoryService(Map<?, ?> parameters) throws Re
...
-                // using default
-            }
-        }
-        
-        if (maximumHttpConnections > 0) {
-            return new RepositoryServiceImpl(uri, idFactory, nameFactory, pathFactory, vFactory, itemInfoCacheSize, maximumHttpConnections);
-        } else {
-            return new RepositoryServiceImpl(uri, idFactory, nameFactory, pathFactory, vFactory, itemInfoCacheSize);
-        }
+        return new RepositoryServiceImpl(uri, idFactory, nameFactory, pathFactory, vFactory, itemInfoCacheSize, ConnectionOptions.fromServiceFactoryParameters("org.apache.jackrabbit.spi2dav", parameters));
```
Comment (kwin) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/Spi2davRepositoryServiceFactory.java]: no dynamic prefix should be used for connection parameters

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java [hunk 5]
```diff
@@ -226,7 +243,9 @@
     /**
      * Default value for the maximum number of connections per host such as
      * configured with {@link PoolingHttpClientConnectionManager#setDefaultMaxPerRoute(int)}.
+     * @deprecated Use {@link ConnectionOptions#MAX_CONNECTIONS_DEFAULT} instead
      */
+    @Deprecated
```
Comment (reschke) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java]: I believe we can just remove this.

**Review by reschke — COMMENTED** [review]
Looks good overall.

File: jackrabbit-spi2dav/pom.xml [hunk 6]
```diff
@@ -1,189 +1,264 @@
...
+            <scope>test</scope>
+        </dependency>
+        <!-- for testing connection via a proxy -->
+        <dependency>
+            <groupId>org.littleshoot</groupId>
+            <artifactId>littleproxy</artifactId>
+            <version>1.1.2</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (reschke) [inline jackrabbit-spi2dav/pom.xml]: Could you please backout the whitespace changes?

**Review by reschke — CHANGES_REQUESTED** [review]

**Comment by kwin** [comment]
Merged in https://github.com/apache/jackrabbit/commit/d545fa178cccf42f60cbc9d28121c7f0a99db3b1.

**Pull request closed without merging**